### DISCUSSION TITLE
Storing access_token on directus side.

### DISCUSSION THREAD

**Discussion by bhagya1101** [directus/directus#16805] [Archive]
Suppose a custom api is created using ' npm init directus-extension ' and I want a particular user with his/her access_token to get their own details (i.e. for /users/me). How do I store the access_token that is generated when user signs in and pass it as an Authorization header in POST method of API  ?

The above process has to be done in directus only.
I am somewhat new to it, so , any help would be appreciated.

**Comment by br41nslug** [comment]
Generally the App handles the access token for you this is not something you should manage yourself within the app. How and where are you calling this custom endpoint?

**Reply by bhagya1101** [reply]
I want to call a custom API which I have named as '/api/users/me-details' which will give me user-details of current user. So, I want to get data of current user from '/users/me' and show it in my api, how should I do it ?

**Reply by paescuj** [reply]
It seems like you want to enrich the data returned on the `/users/me` endpoint, right? Instead of creating a separate endpoint for this purpose only, you might be able to use a [Custom API Hook](https://docs.directus.io/extensions/hooks.html#custom-api-hooks) to intercept and enrich the payload.
If you really want / have to use a separate [endpoint](https://docs.directus.io/extensions/endpoints.html), you might be able to get the token from `req` (not entirely sure about this) and then use it to call `/users/me`.
